dist/tools/ci/static_tests.sh

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env bash
+#
+# Copyright (C) 2020 Kaspar Schleiser <kaspar@schleiser.de>
+#               2020 Inria
+#               2020 Freie Universität Berlin
+#               2015 Philipp Rosenkranz <philipp.rosenkranz@fu-berlin.de>
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+function print_result {
+    local RED="\033[0;31m"
+    local GREEN="\033[0;32m"
+    local NO_COLOUR="\033[0m"
+
+    if (( "$1" == 0 )); then
+        echo -e "${GREEN}✓$NO_COLOUR"
+    else
+        echo -e "${RED}x$NO_COLOUR"
+    fi
+}
+
+set_result() {
+    NEW_RESULT=$1
+
+    if (( $NEW_RESULT != 0))
+    then
+        RESULT=$NEW_RESULT
+    fi
+}
+
+function run {
+    echo -n "Running '$@' "
+    OUT=$($@ 2>&1)
```

Comment:
Double quote array expansions to avoid re-splitting elements. [[SC2068]](https://github.com/koalaman/shellcheck/wiki/SC2068)

Reply:
This is executed in shell, wouldn't `"foo bar"` try to execute the binary `"foo bar"` instead of `foo` with parameter `bar`?

Reply:
no, `"$@"` is handled correctly. fixed

Reply:
According to the [manual](https://www.gnu.org/software/bash/manual/html_node/Special-Parameters.html) it should expand to a quoted string of every element.